src/Geta.NotFoundHandler/Core/Redirects/CustomRedirectCollection.cs

```diff
@@ -19,14 +19,9 @@ public class CustomRedirectCollection : IEnumerable<CustomRedirect>
     private readonly IEnumerable<INotFoundHandler> _providers = new List<INotFoundHandler>();
 
     /// <summary>
-    /// Hashtable for quick lookup of old urls
+    /// URLs sorted Z-A.
     /// </summary>
-    private readonly Dictionary<string, CustomRedirect> _quickLookupTable = new(StringComparer.OrdinalIgnoreCase);
-
-    /// <summary>
-    /// Cache of URLs sorted ZA for look up of partially matched URLs
-    /// </summary>
-    private KeyValuePair<string, CustomRedirect>[] _redirectsZACache;
+    private readonly SortedDictionary<string, CustomRedirect> _redirectsZA = new(new ReverseStringComparer());
```

Comment:
Switching from Dictionary (O(1) lookup) to SortedDictionary (O(log n)) might actually hurt performance, especially since:
* initialization would take more time and be more processing intensive (as it would have to remain sorted at all times)
* the main use case involves frequent lookups
* the performance would be dependent on the collection size

https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2?view=net-9.0
https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.sorteddictionary-2?view=net-9.0